### PR TITLE
Add Metric_Practice

### PR DESCRIPTION
**Pull request by Khaleeh**

### CODE REVIEW

File: Metric_practice.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import numpy as np
```
Comment (FanwangM) [inline Metric_practice.py:1]: We usually don't use capitalized letters in python file names.

File: Metric_practice.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# mock feature matrix
+feature = np.array([  [4, 2, 6],
+                      [4, 9, 6],
+                      [2, 5, 0],
+                      [2, 0, 9],
+                      [5, 3, 0]])
```
Comment (FanwangM) [inline Metric_practice.py]: Extra space should be removed.

File: Metric_practice.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    x : first vector
+    y : second vector
+    -------
+    Returns
+    -------
+    distance : euclidan distance between x and y
+    """
+    intermdiate = 0
+    for i in range(0, len(x)):
+        intermdiate += ((x[i] - y[i])**2)
```
Comment (FanwangM) [inline Metric_practice.py]: should be 
```python
intermdiate += ((x[i] - y[i])**2)
```

File: Metric_practice.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    ----------
+    Parameters
+    ----------
+    x : first vector
+    y : second vector
+    -------
+    Returns
+    -------
+    distance : euclidan distance between x and y
+    """
```
Comment (FanwangM) [inline Metric_practice.py]: Not following `numpy` documentation style.

File: Metric_practice.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    matrix : given matrix
+    -------
+    Returns
+    -------
+    perwise_distance : the condensed pairwise distance
+    """
+    per_dist = []
+    for i in range(0,(len(matrix)-1)):
+        for j in range(i + 1,(len(matrix))):
+            per_dist.append(euclidean_distance(matrix[i],matrix[j]))
```
Comment (FanwangM) [inline Metric_practice.py]: We usually put a space after the operators.

File: Metric_practice.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    return per_dist
+
+print(pairwaise_distance(feature))
```
Comment (FanwangM) [inline Metric_practice.py]: If this file is intended to be a command line tool, should wrap with 

```python
if __name__ == "__main__":
    print(pairwaise_distance(feature))
```

File: Metric_practice.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    ----------
+    x : first vector
+    y : second vector
+    -------
+    Returns
+    -------
+    distance : euclidan distance between x and y
+    """
+    intermdiate = 0
+    for i in range(0, len(x)):
```
Comment (FanwangM) [inline Metric_practice.py]: Better to use `np.shape(x)` or `x.shape[]`?

File: Metric_practice.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    Parameters
+    ----------
+    matrix : given matrix
+    -------
+    Returns
+    -------
+    perwise_distance : the condensed pairwise distance
+    """
+    per_dist = []
+    for i in range(0,(len(matrix)-1)):
```
Comment (FanwangM) [inline Metric_practice.py]: Can be simplified to ```for j in range(i + 1, (len(matrix))):```.

**Review by FanwangM — COMMENTED** [review]
It looks good in general, but some fixes are required.

File: Metric_practice.py (hunk 8 above)
Comment (Khaleeh) [reply Metric_practice.py]: Most Issues have been addressed in 1733172.

File: Metric_practice.py [hunk 9]
```diff
@@ -1,50 +1,56 @@
...
+    """Pairwise distance between vectors
+
+    Computes the pairwise distance between each vector
+    in a given array
+    
     Parameters
     ----------
-    matrix : given matrix
-    -------
+    Z : ndarray
```
Comment (FanwangM) [inline Metric_practice.py]: We usually don't have a capitalized letter as an argument name. Low case is a good practice.

File: Metric_practice.py [hunk 10]
```diff
@@ -1,50 +1,56 @@
...
-    """
-    Computes the pairwise distance between each vectore
-    in a given matrix
-    ----------
+def pairwise_distance(Z):
+    """Pairwise distance between vectors
+
+    Computes the pairwise distance between each vector
+    in a given array
+
```
Comment (FanwangM) [inline Metric_practice.py]: Check `numpy` docstring style please.

```
"""Computes the pairwise distance between each vector in a given array.
"""
```

**Review by FanwangM — CHANGES_REQUESTED** [review]

**Pull request merged**